main.js

```diff
@@ -2,5 +2,127 @@
  * Created by Jenya on 18.04.15.
  */
 
+/// At first I'm creating general BeingClass. It will be the top of all being classes.
+function BeingClass(clas) {
+    this.clas = clas;
+    this.movementType = move.run;
+    this.legs = 0
+}
+
+// Than I'm adding move function to all beings.
+BeingClass.prototype.makeMove = function () {
+    this.movementType.move(this.name);
+    return this;
+};
+
+BeingClass.prototype.setLegsAmount = function (legsAmount) {
+    this.legs = legsAmount;
+    return this;
+};
+
+//Making a method to print object's properties
+BeingClass.prototype.logName = function () {
+    console.log("This.class: " + this.clas + "; this.name: " + this.name + "; this.legs= " + this.legs);
+    return this;
+};
+
+//Now I'm setting movement methods, strategy and type switch - BEGIN
+BeingClass.prototype.setMovementType = function (newMovementType) {
+    this.movementType = newMovementType;
+    return this;
+};
+
+var Movement = function (func) {
+    this.move = func;
+};
+
+var move = {};
+
+move.jump = new Movement(function (name) {
+    console.log(name + " is jumping!")
+});
+
+move.run = new Movement(function (name) {
+    console.log(name + " is running!");
+});
+move.fly = new Movement(function (name) {
+    console.log(name + " is flying!")
+});
+
+//setting movement methods, strategy and type switch - END
+
+
+// ---- please skip these comments
+/// setting speak methods and strategy
```

Comment:
As you wish... By the way, let's make now next iteration:
1) merge this code to master
2) create new branch where you will put each file into its own file
3) for now let the scripts be added to main html one by one in script tags
4) then you will use require.js to combine them
5) then you will use browserify to combine them